### PR TITLE
Cs 1214 relay email address update approvals

### PR DESCRIPTION
**Pull request by Pauljgraham**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeCommitments/Data/Models/Apprentice.cs [hunk 1]
```diff
@@ -70,6 +71,7 @@ internal void UpdateEmail(MailAddress newEmail)
             if (newEmail.Address == Email.Address) return;
```
[the remark this answers is not in the document]
Comment (benagain) [reply src/SFA.DAS.ApprenticeCommitments/Data/Models/Apprentice.cs:71]: I like this PR, nice and clean and compact.  

I'm not a fan of the `ApprenticeshipPatcher` class - it seems to me that we would be better to use the DTO and mapping concepts we have already built up.

File: src/SFA.DAS.ApprenticeCommitments/Data/IApprenticeContext.cs [hunk 2]
```diff
@@ -1,29 +1,41 @@
...
+
+        internal async Task<Apprentice> GetByIdAndIncludeApprenticeships(Guid apprenticeId)
+            => await FindAndIncludeApprenticeships(apprenticeId)
+               ?? throw new DomainException(
+                   $"Apprentice {apprenticeId} not found");
+
+        internal async Task<Apprentice?> FindAndIncludeApprenticeships(Guid apprenticeId)
+            => await Entities
+                .Include(e => e.Apprenticeships)
+                .ThenInclude(e => e.Revisions)
```
[the remark this answers is not in the document]
Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticeCommitments/Data/IApprenticeContext.cs:38]: it's needed as we have to get the latest revision to pick up the CommitmentApprenticeshipId value. If we move that value to the apprenticeship record it wouldn't be needed

File: src/SFA.DAS.ApprenticeCommitments/Data/ApprenticePatcher.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Net.Mail;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.ApprenticeCommitments.Data.Models;
+
+namespace SFA.DAS.ApprenticeCommitments.Data
+{
+    public sealed class ApprenticePatcher
```
[the remark this answers is not in the document]
Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticeCommitments/Data/ApprenticePatcher.cs:8]: The reason to put the patcher in is to abstract the apprentice domain object away from Patch operations. I feel we loose control with an open Patch directly to the domain object. This object allows us to protect the domain object from intrusive calls, log the operations which have occurred and call alterative methods to the direct property set.

File: src/SFA.DAS.ApprenticeCommitments/Data/Models/Apprentice.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticeCommitments/Data/Models/Apprentice.cs:71]: Lets discuss

File: src/SFA.DAS.ApprenticeCommitments/Data/IApprenticeContext.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (benagain) [reply src/SFA.DAS.ApprenticeCommitments/Data/IApprenticeContext.cs:38]: Ah, right.  Yes I see it now

**Pull request closed without merging**